aten/src/ATen/native/sparse/SparseUnaryOps.cpp

```diff
@@ -200,20 +201,38 @@ Tensor isinf_sparse_meta(const Tensor& self) {
   TORCH_CHECK_NOT_IMPLEMENTED(0, "nyi isinf for SparseMeta");
 }
 
+Tensor zeros_like_sparse(const Tensor& t) {
```

Comment:
@amjames - didn't we add a few helper functions for behavior like this already?

Reply:
We might have it for CSR, but then just `at::zeros_like` is sufficient for compressed formats. `zeros_like` for sparse COO just returns an empty tensor. But I could add to the sparse utils header indeed.

Reply:
I have found that `zeros_with_indices_like` is far more useful than `zeros_like` which returns a tensor with zero-sized indices/values. A design suggestion:
- `r = torch.empty_like(sparse)` returns a sparse tensor `r` such that
    - `r.layout == sparse.layout`
    - `r.sparse_dim() == sparse.sparse_dim()`
    - `r.dense_dim() == sparse.dense_dim()` (not relevant for COO but it is relevant for sparse compressed tensors)
    - `r._nnz() == 0`.
- `r = torch.zeros_like(sparse)` returns a sparse tensor `r` such that
    - `r.layout == sparse.layout`
    - `r.sparse_dim() == sparse.sparse_dim()`
    - `r.dense_dim() == sparse.dense_dim()` (not relevant for COO but it is relevant for sparse compressed tensors)
    - `r.indices() == sparse.indices()`

^ @amjames @cpuhrsch

Reply:
I am with @pearu on this one.